pkg/lvmd/cmd/root.go

```diff
@@ -97,31 +99,51 @@ func subMain(ctx context.Context) error {
 	proto.RegisterVGServiceServer(grpcServer, vgService)
 	proto.RegisterLVServiceServer(grpcServer, lvmd.NewLVService(dcm, ocm, notifier))
 	grpc_health_v1.RegisterHealthServer(grpcServer, lvmd.NewHealthService())
-	well.Go(func(ctx context.Context) error {
-		return grpcServer.Serve(lis)
-	})
-	well.Go(func(ctx context.Context) error {
-		<-ctx.Done()
+
+	var wg sync.WaitGroup
+	errCh := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := grpcServer.Serve(lis); err != nil {
```

Comment:
`well.Go` terminate all go routine when any go routine is finished. This implementation, the program will not stop even if grpcServer.Serve get terminated before any other.

Reply:
That's right. I'll fix my code.

Reply:
I fixed my code, and every goroutine now calls `cancel()` at the end to stop other goroutines.